### PR TITLE
Feat/supabase

### PR DESCRIPTION
**Pull request by Ashutosh-Bhadauriya**
Adds supabase guide along with the starter repo for it

### CODE REVIEW

File: guides/databases/supabase.mdx [hunk 1]
```diff
@@ -0,0 +1,231 @@
+---
+sidebarTitle: Supabase
+title: Using Hanko with Supabase
+description: This guide showcases how to add new user info, like user ID and email to the Supabase Database, when they sign up. Jump to [Nextjs guide](/quickstarts/fullstack/next) to learn how to add Hanko Auth in your Next App.
+---
+
+We're using Next.js in this guide, but don't worry, you can apply the same concepts into other frameworks or languages. We're working on adding examples with different frameworks, so stay tuned!
```
Comment (lfleischmann) [inline guides/databases/supabase.mdx]:
```suggestion
We're using Next.js in this guide, but don't worry, you can apply the same concepts to other frameworks or languages. We're working on adding examples with different frameworks, so stay tuned!
```

File: guides/databases/supabase.mdx [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+    ```sh .env.local
+    NEXT_PUBLIC_HANKO_API_URL=https://f4****-4802-49ad-8e0b-3d3****ab32.hanko.io
+
+    NEXT_PUBLIC_SUPABASE_PROJECT_URL=YOUR_SUPABASE_PROJECT_URL
+    NEXT_PUBLIC_SUPABASE_SERVICE_ROLE_KEY=YOUR_SUPABASE_SERVICE_ROLE_KEY
+    ```
+
+  </Step>
+  <Step title="Install Supabase JavaScript Client">
+    Assuming you've already setup Hanko Auth, you'll need to install supabase to your to your app.
```
Comment (lfleischmann) [inline guides/databases/supabase.mdx]:
```suggestion
    Assuming you've already set up Hanko Auth, you'll need to install supabase to your app.
```

File: guides/databases/supabase.mdx [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+        ...user,
+        id: supauser?.id,
+        };
+
+        return new NextResponse(JSON.stringify(user_data), { status: 200 });
+    } catch (error) {
+        console.error("[CREATEUSER_ERROR]", error);
+        return new NextResponse("Internal Error", { status: 500 });
+    }
+    }
```
Comment (lfleischmann) [inline guides/databases/supabase.mdx]: Indentation/Formatting for this code block is a bit off in a couple of places, looks weird when rendered.

File: guides/databases/supabase.mdx [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+    ```
+
+  </Step>
+  <Step title="Use the API in HankoAuth component">
+    Use the API to add a new user to 'users' table, as soon as user is authenticated and `onAuthFlowCompleted` is triggered.
```
Comment (lfleischmann) [inline guides/databases/supabase.mdx]:
```suggestion
    Use the API to add a new user to the 'users' table, as soon as user is authenticated and `onAuthFlowCompleted` is triggered.
```

File: guides/databases/supabase.mdx [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+    );
+
+    useEffect(() => {
+        register(hankoApi).catch((error) => {
+        // handle error
+        });
+    }, []);
+
+    return <hanko-auth />;
+    }
```
Comment (lfleischmann) [inline guides/databases/supabase.mdx]: Indentation/Formatting off in some places.

File: guides/databases/supabase.mdx [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+      <img
+        src="/images/supabase/create-new-table.png"
+        alt="create new project"
+        width="500"
+        style={{ borderRadius: "0.5rem" }}
+      />
+    </Frame>
+  </Step>
+  <Step title="Get URL and service_role key from Supabase and add it to your project">
+    Now in the project dashboard, select 'Settings' from the left sidebar and then click on 'API'. Copy the 'URL' and 'service_role' key and paste them in your `.env.local` file.
```
Comment (lfleischmann) [inline guides/databases/supabase.mdx]:
```suggestion
    Now in the project dashboard, select 'Project Settings' from the left sidebar and then click on 'API'. Copy the 'URL' and 'service_role' key and paste them in your `.env.local` file.
```

File: guides/databases/supabase.mdx [hunk 7]
```diff
@@ -0,0 +1,231 @@
...
+    ```sh .env.local
+    NEXT_PUBLIC_HANKO_API_URL=https://f4****-4802-49ad-8e0b-3d3****ab32.hanko.io
+
+    NEXT_PUBLIC_SUPABASE_PROJECT_URL=YOUR_SUPABASE_PROJECT_URL
+    NEXT_PUBLIC_SUPABASE_SERVICE_ROLE_KEY=YOUR_SUPABASE_SERVICE_ROLE_KEY
```
Comment (lfleischmann) [inline guides/databases/supabase.mdx]: Is this the one in the `Project API keys` panel? The one marked with the red `secret` badge? The panel tells me to never share this publicly. Should this really be `NEXT_PUBLIC` then?

File: guides/databases/supabase.mdx [hunk 8]
```diff
@@ -0,0 +1,231 @@
...
+            };
+            await fetchData();
+            redirectAfterLogin();
+        }),
+        [hanko, redirectAfterLogin]
+    );
+
+    useEffect(() => {
+        register(hankoApi).catch((error) => {
+        // handle error
```
Comment (lfleischmann) [inline guides/databases/supabase.mdx]: Why say "handle error" via comment here, but above errors are `console.error`-ed? I think it's best if things like this are consistent, so either `console.error` everywhere or comments everywhere.

File: guides/databases/supabase.mdx [hunk 9]
```diff
@@ -0,0 +1,231 @@
...
+      <img
+        src="/images/supabase/create-new-project.png"
+        alt="create new project"
+        width="500"
+        style={{ borderRadius: "0.5rem" }}
+      />
+    </Frame>
+  </Step>
+  <Step title="Create users table">
+    Now, in the project dashboard, choose 'Database' and create a new 'users' table.
```
Comment (lfleischmann) [inline guides/databases/supabase.mdx]:
```suggestion
    Now, on the project dashboard, choose 'Database' and create a new 'users' table.
```

File: guides/databases/supabase.mdx [hunk 10]
```diff
@@ -0,0 +1,231 @@
+---
+sidebarTitle: Supabase
+title: Using Hanko with Supabase
+description: This guide showcases how to add new user info, like user ID and email to the Supabase Database, when they sign up. Jump to [Nextjs guide](/quickstarts/fullstack/next) to learn how to add Hanko Auth in your Next App.
```
Comment (lfleischmann) [inline guides/databases/supabase.mdx]:
```suggestion
description: This guide showcases how to add new user info, like user ID and email to the Supabase database when  users sign up with Hanko. Jump to [Nextjs guide](/quickstarts/fullstack/next) to learn how to add Hanko Auth in your Next App.
```

File: guides/databases/supabase.mdx [hunk 11]
```diff
@@ -0,0 +1,231 @@
...
+        alt="create new project"
+        width="500"
+        style={{ borderRadius: "0.5rem" }}
+      />
+    </Frame>
+  </Step>
+  <Step title="Create users table">
+    Now, in the project dashboard, choose 'Database' and create a new 'users' table.
+    We'll be using this table to add new user info, on signup. Along with the
+    default 'id' column, add 'user_id' and 'email' columns.
```
Comment (lfleischmann) [inline guides/databases/supabase.mdx]:
```suggestion
    We'll be using this table to add new user info on signup. In addition to the
    default 'id' column, add 'user_id' and 'email' columns.
```

File: guides/databases/supabase.mdx (hunk 7 above)
Comment (Ashutosh-Bhadauriya) [reply guides/databases/supabase.mdx]: Yeah, sorry. Actually, we don't need NEXT_PUBLIC in both project url and service role key

**Comment by Ashutosh-Bhadauriya** [comment]
Thanks @lfleischmann

**Review by lfleischmann — APPROVED** [review]

**Pull request merged**